### PR TITLE
Fix makeUrl for Fastly

### PR DESCRIPTION
**Pull request by jimbo**
## Description

Fix `makeUrl` for Fastly image optimization.

## Related Issue

- #1038

## Motivation and Context

URLs from GraphQL are now absolute, and they include the media path logic we currently have in `makeUrl`. So we need to remove this logic and rewrite what remains to be compatible with absolute URLs. Also, we need to enable Fastly image optimization in certain environments.

## Verification

- Ensure category images are loading on the home page
- Ensure you can soft navigate (click) from home page to a category page
- Ensure product images are loading in the category gallery
- Ensure you can soft navigate from a category page to a product page
- Ensure product images are loading
- Ensure you can soft navigate to the home page via the header logo

## How Has This Been Tested?

- UAT with the above method, image optimization enabled
- UAT with the above method, image optimization disabled, Fastly enabled
- `yarn test`

### CODE REVIEW

File: packages/venia-concept/src/util/makeUrl.js [hunk 1]
```diff
@@ -1,64 +1,77 @@
...
-    `${type}%%${url}%%${width}`;
+const formatUrl = (url = '', { useFastly, type, width } = {}) =>
+    makeOptimizedUrl(
+        prependMediaPath(url, type),
+        width,
+        useFastly || process.env.USE_FASTLY
+    );
 
-export default memoize(formatUrl, storeUrlKey);
+export default formatUrl;
```
Comment (jimbo) [inline packages/venia-concept/src/util/makeUrl.js]: Could add memoization back at some point, but it should take `useFastly` into account.

File: packages/venia-concept/src/util/makeUrl.js [hunk 2]
```diff
@@ -1,64 +1,56 @@
...
+        process.env.MAGENTO_BACKEND_MEDIA_PATH_CATEGORY ||
+            '/media/catalog/category'
+    );
+
+const resizeBase = joinUrls(
+    process.env.IMAGE_SERVICE_PATH || '/img/',
+    '/resize/'
+);
+
+const makeOptimizedUrl = (path, { type, width } = {}) => {
```
Comment (sirugh) [inline packages/venia-concept/src/util/makeUrl.js]: A jsdoc style comment explaining the functionality would be really helpful here!

File: packages/venia-concept/src/util/makeUrl.js [hunk 3]
```diff
@@ -1,64 +1,56 @@
...
+        'image-category',
+        process.env.MAGENTO_BACKEND_MEDIA_PATH_CATEGORY ||
+            '/media/catalog/category'
+    );
+
+const resizeBase = joinUrls(
+    process.env.IMAGE_SERVICE_PATH || '/img/',
+    '/resize/'
+);
+
```
Comment (sirugh) [inline packages/venia-concept/src/util/makeUrl.js]:
```suggestion
/**
 * Creates an "optimized" url for a provided absolute or relative url based on
 * requested media type and width.
 *
 * If a `type` is provided the `path` will be joined with the associated media
 * base.
 *  - "/media/catalog/product/some/path/to/img.jpg"
 *
 * If a `width` is provided a "resize url" is returned using the desired width
 * and original media url.
 *  - /img/resize/640?url=%2Fmedia%2Fcatalog%2Fproduct%2Fsome%2Fpath%2Fto%2F/image.jpg
 *
 * If only `path` is provided it is returned unaltered.
 *
 * @param {string} path - absolute or relative url to resource.
 * @param {Object} props - properties describing desired optimizations
 * @param {string} props.type - "image-product" or "image-category"
 * @param {number} props.width - the desired resize width of the image
 */
```

File: packages/venia-concept/src/util/makeUrl.js (hunk 1 above)
Comment (zetlen) [reply packages/venia-concept/src/util/makeUrl.js]: Of course, it was the memoization. Should have figured that out.

**Pull request merged**